packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -559,45 +577,60 @@ class HeroController extends NavigatorObserver {
   }
 
   @override
-  void didStartUserGesture() {
-    _questsEnabled = false;
-  }
-
-  @override
-  void didStopUserGesture() {
-    _questsEnabled = true;
+  void didStartUserGesture(Route<dynamic> route, Route<dynamic> previousRoute) {
+    assert(navigator != null);
+    assert(route != null);
```

Comment:
It's OK if previousRoute is null?

Reply:
It is. And it should be ok for _maybeStartHeroTransition to drop it in that case too.